### PR TITLE
[Integration Test] - Automation test for few modules 

### PR DESCRIPTION
**Pull request by ShivamSh3**
# Description
Integration test for few below modules

- redfish_event_subscription
- redfish_firmware
- redfish_firmware_rollback
- redfish_powerstate
- idrac_redfish_storage_controller

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

**Comment by ShivamSh3** [comment]
image

**Comment by ShivamSh3** [comment]
image

**Comment by ShivamSh3** [comment]
image

**Comment by ShivamSh3** [comment]
image

**Comment by ShivamSh3** [comment]
image

**Review by sachin-apa — DISMISSED** [review]

**Review by ABHISHEK-SINHA10 — DISMISSED** [review]

**Review by sachin-apa — DISMISSED** [review]

**Comment by ShivamSh3** [comment]
image

**Comment by ShivamSh3** [comment]
image

**Review by sachin-apa — DISMISSED** [review]

**Comment by ShivamSh3** [comment]
image

**Comment by ShivamSh3** [comment]
image

**Review by singh-lovepreet1 — APPROVED** [review]

File: tests/integration/inventory.networking [hunk 1]
```diff
@@ -7,4 +7,8 @@ ansible_host=localhost
 ansible_connection=local
 ansible_become_pass=password
 idrac_reset_default_username=default_username
-idrac_reset_default_password=default_password
\ No newline at end of file
+idrac_reset_default_password=default_password
+http_share=default_http_share
```
Comment (rajshekarp87) [inline tests/integration/inventory.networking]:
```suggestion
http_share=http_share_ip
```

File: tests/integration/inventory.networking [hunk 2]
```diff
@@ -7,4 +7,8 @@ ansible_host=localhost
...
+http_share_dir_path=default_http_share_path
```
Comment (rajshekarp87) [inline tests/integration/inventory.networking]:
```suggestion
http_share_dir_path=http_share_path
```

File: tests/integration/inventory.networking [hunk 3]
```diff
@@ -7,4 +7,8 @@ ansible_host=localhost
...
+idrac_firmware_dup_file=default_idrac_fw_file
```
Comment (rajshekarp87) [inline tests/integration/inventory.networking]:
```suggestion
idrac_firmware_dup_file=idrac_firmware_dup_file
```

File: tests/integration/inventory.networking [hunk 4]
```diff
@@ -7,4 +7,8 @@ ansible_host=localhost
...
+bios_firmware_dup_file=default_bios_fw_file
```
Comment (rajshekarp87) [inline tests/integration/inventory.networking]:
```suggestion
bios_firmware_dup_file=bios_firmware_dup_file
```

**Review by Saksham-Nautiyal — APPROVED** [review]

File: tests/integration/inventory.networking (hunk 2 above)
Comment (ShivamSh3) [reply tests/integration/inventory.networking]: done

File: tests/integration/inventory.networking (hunk 4 above)
Comment (ShivamSh3) [reply tests/integration/inventory.networking]: done

File: tests/integration/inventory.networking (hunk 3 above)
Comment (ShivamSh3) [reply tests/integration/inventory.networking]: done

File: tests/integration/inventory.networking (hunk 1 above)
Comment (ShivamSh3) [reply tests/integration/inventory.networking]: done

**Review by ABHISHEK-SINHA10 — APPROVED** [review]

**Review by rajshekarp87 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Pull request merged**